pkg/workceptor/kubernetes.go

```diff
@@ -661,7 +669,37 @@ func (kw *kubeUnit) Start() error {
 
 // Restart resumes monitoring a job after a Receptor restart
 func (kw *kubeUnit) Restart() error {
-	return fmt.Errorf("restart of Kubernetes pod not implemented")
+	status := kw.Status()
+	ked := status.ExtraData.(*kubeExtraData)
+	if IsComplete(status.State) {
+		return nil
+	}
+	isTCP := kw.streamMethod == "tcp"
+	if status.State == WorkStateRunning && !isTCP {
+		return kw.startOrRestart()
+	}
+	// Work unit is in Pending state
+	if kw.deletePodOnRestart {
+		err := kw.connectToKube()
+		if err != nil {
+			logger.Warning("Pod %s could not be deleted: %s", ked.PodName, err.Error())
+		} else {
+			err := kw.clientset.CoreV1().Pods(ked.KubeNamespace).Delete(context.Background(), ked.PodName, metav1.DeleteOptions{})
```

Comment:
I noticed in `Cancel` we put this delete in a go routine, but I didn't have any issues with blocking on this api call. Should I also put this into a go routine?

Reply:
As discussed, there doesn't seem to be a reason for either to be in separate goroutines.